linked_list/linked_list.py

```diff
@@ -86,10 +153,24 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def reverse(self):

```

Comment:
👀 Again, you've implemented this function correctly for a doubly linked list, but your Linked List class only represents a singly linked list!